mcu/src/mcu/uart_reader.py

```diff
@@ -0,0 +1,455 @@
+    voltage_mV = signal * VDD / VAL_MAX_ADC * 1e3
+
+    # Calculate FFT
+    n = len(voltage_mV)
+    fft_values = np.fft.fft(voltage_mV)
+    fft_values = np.abs(fft_values)[: n // 2]  # Magnitude of the FFT
+    freqs = np.fft.fftfreq(n, 1 / sampling_rate)[: n // 2]  # Associated frequencies
+
+    # Exclude the DC component (frequency 0)
+    freqs = freqs[1:]
+    fft_values = fft_values[1:]
+
+    # Create subplot with 2 graphs (time-domain + FFT)
+    fig = make_subplots(
+        rows=2, cols=1, subplot_titles=("Time-Domain Signal", "Signal FFT")
+    )
+
+    fig.update_layout(
+        title="Real-Time Acquisition",
+        xaxis_title="Time (s)",
+        yaxis_title="Voltage (mV)",
+    )
+    fig.update_yaxes(title_text="Voltage (mV)", row=1, col=1)
+    fig.update_yaxes(title_text="FFT Magnitude", row=2, col=1)
+    fig.update_xaxes(title_text="Time (s)", row=1, col=1)
+    fig.update_xaxes(title_text="Frequency (Hz)", row=2, col=1)
+
+    # Plot time-domain signal
+    fig.add_trace(
+        go.Scatter(
+            x=times, y=voltage_mV, mode="lines", name=f"Acquisition #{msg_counter}"
+        ),
+        row=1,
+        col=1,
+    )
+    # Plot FFT without DC component
+    fig.add_trace(
+        go.Scatter(x=freqs, y=fft_values, mode="lines", name=f"FFT #{msg_counter}"),
+        row=2,
+        col=1,
+    )
+
+    # Show plots
+    pio.show(fig, renderer="browser")
+
+    # Generate audio file
+    generate_audio(signal, f"acq-{msg_counter}")
+
+    msg_counter += 1
+
+
+def print_and_save_data(data):
+    if LOG_ENABLED:
+        with open("uart_reader.log", "w") as f:
+            f.write(data)
+    print(data)
+
+
+def read_serial(port, baudrate=DEFAULT_BAUDRATE):
+    """Reads the serial data and processes it for the console mode."""
+    try:
+        with serial.Serial(port, baudrate) as ser:
+            print_and_save_data(f"Connected to {port} with baudrate {baudrate}")
+            while not stop_event.is_set():
+                line = ser.readline().decode("ascii").strip()
+                if line.startswith(PRINT_PREFIX):
+                    buffer = bytes.fromhex(line[len(PRINT_PREFIX) :])
+                    dt = np.dtype(np.uint16)
+                    dt = dt.newbyteorder("<")
+                    buffer_array = np.frombuffer(buffer, dtype=dt)
+                    print_and_save_data(f"Aquisition Number : {msg_counter}")
+                    print_and_save_data(
+                        f">Aquisition Memory: {len(line) - len(PRINT_PREFIX)} bytes"
+                    )
+                    print_and_save_data(
+                        f">Aquisition Size  : {len(buffer_array)} samples"
+                    )
+                    plot_signal_and_fft(buffer_array, FREQ_SAMPLING)
+                else:
+                    print_and_save_data(f"{line}")
+    except Exception as e:
+        print_and_save_data(f"Error: {e}")
+        sys.exit(1)
+    finally:
+        print_and_save_data("Disconnected")
```

Comment:
Just remove the plain try/catch, it makes little sense to catch "any exception" (this is also considered bad practice in Python), and let the error propagate to the program, this will eventually give more information to the user.